src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,168 @@
+package duke;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+import duke.task.ToDoTask;
+import duke.task.Task;
+import duke.task.EventTask;
+import duke.task.DeadlineTask;
+
+/**
+ * Class to abstract the storing of commands
+ */
+public class Storage {
+
+    File storageFile;
+
+    /**
+     * Constructor for the Storage Class
+     *
+     * @param filePath The path of the file
+     */
+    public Storage(String filePath) {
+        if (filePath != null) {
+            storageFile = new File(filePath);
+        }
+    }
+
+    /**
+     * Method to save the Task List in the File
+     *
+     * @param tasks The Task List of the Duke
+     */
+    public void saveTaskList(TaskList tasks) {
+        UI ui = new UI();
+        try {
+            File dir = new File("data");
+            boolean isDeleted = false, isCreated = false, isDirectory = dir.mkdirs();
+            File data = new File(dir, "duke.txt");
+            if (isDirectory && data.exists()) {
+                isDeleted = data.delete();
+            }
+            if(isDeleted) {
+                isCreated = data.createNewFile();
+            }
+            if (isCreated) {
+                FileWriter storeData = new FileWriter(data);
+                Task task;
+                for (int i = 0; i < tasks.getTaskListLength(); i++) {
+                    task = tasks.getTask(i);
+                    if (task instanceof DeadlineTask) {
+                        DeadlineTask deadline = (DeadlineTask) task;
+                        storeData.write(deadline.writeToFile());
+                    } else if (task instanceof EventTask) {
+                        EventTask event = (EventTask) task;
+                        storeData.write(event.writeToFile());
+                    } else if (task instanceof ToDoTask) {
+                        ToDoTask todo = (ToDoTask) task;
+                        storeData.write(todo.writeToFile());
+                    } else {
+                        ui.showLine();
+                        ui.showError("Error Occurred While Storing Data");
+                        ui.showLine();
+                    }
+                }
+                storeData.close();
+            }
+        } catch (IOException e) {
+            File deletedFile = new File("data/duke.txt");
+            boolean isDeleted = deletedFile.delete(), shouldShow = false;
```

Comment:
Maybe canShow is more appropriate than shouldShow for a boolean?

Reply:
Both Seem similar so will discuss with tutors and accordingly change. Thank you for pointing it out.